src/ConnectComponents.tsx

```diff
@@ -13,12 +13,33 @@ ConnectComponentsContext.displayName = 'ConnectComponents';
 export const ConnectComponentsProvider = ({
   connectInstance,
   children,
+  initParams,
 }: {
   connectInstance: connectJs.StripeConnectInstance;
   children: any;
+  initParams?: connectJs.IStripeConnectInitParams | undefined;
 }): JSX.Element => {
+  const [storedConnectInstance, setStoredConnectInstance] =
+    React.useState<connectJs.StripeConnectInstance>(connectInstance);
+
+  if (storedConnectInstance === null && !initParams) {
+    throw new Error();
+  }
+
+  React.useEffect(() => {
+    if (connectInstance === null && initParams) {
+      const createInstance = async () => {
+        const connectWrapper = await connectJs.loadConnect();
```

Comment:
Because of the async nature of this function, there is a race condition where "the last connect instance to load wins". The last connect instance may not be the right one (this is similar to the race condition our earlier implementation with the `promise` had)

Reply:
I don't think this will happen in practice (I think*, since the only `async` function here is `loadConnect`, and that should only happen once) however it is possible